rust_src/src/decompress.rs

```diff
@@ -0,0 +1,127 @@
+//! Interface to zlib.
+use std::cmp::min;
+use std::io::prelude::Read;
+use std::slice;
+
+use buffers::validate_region;
+use flate2::read::{DeflateDecoder, GzDecoder, ZlibDecoder};
+use lisp::LispObject;
+use lisp::defsubr;
+use remacs_macros::lisp_fn;
+use remacs_sys::{buf_charpos_to_bytepos, del_range, insert_from_gap, make_gap, maybe_quit,
+                 move_gap_both};
+use threads::ThreadState;
+
+/// Return t if zlib decompression is available in this instance of Emacs.
+#[lisp_fn]
+pub fn zlib_available_p() -> bool {
+    true
+}
+
+fn create_buffer_decoder<'a>(buffer: &'a [u8]) -> Box<Read + 'a> {
+    let magic_number = buffer[0];
+
+    match magic_number {
+        // Zlib
+        0x78 => Box::new(ZlibDecoder::new(buffer)),
+        // Gzlib
+        0x1F => Box::new(GzDecoder::new(buffer)),
```

Comment:
I think the comment reads better this way.

`// Assume the data is raw if neither zlib nor gzip header can be found.`

Reply:
Fixed it. Thanks.